nemo/core/optim/lr_scheduler.py

```diff
@@ -496,14 +496,22 @@ def prepare_lr_scheduler(
     elif 'iters_per_batch' in scheduler_config:
         # Compute effective max_steps if iters_per_batch is provided
         if train_dataloader is None:
-            raise ValueError(
-                'As `iters_per_batch` is provided, it is required to pass the train dataloader in order '
-                'to compute effective maximum number of steps'
+            logging.warning(
+                'As `iters_per_batch` is provided/computed, it is required to pass the train dataloader in order\n'
+                'to compute effective maximum number of steps.\n'
+                'Scheduler will not be instantiated !'
             )
+            return None
```

Comment:
What's the reasoning for this change?

Reply:
Trainer.test() calls optimizer + scheduler setup (it calls the internal fit call with different parameters). Since the fit call has the configire_optimizers() call, the scheduler is attempted to be built. 

Of the model is tested right after training, then there's no issues as the train ds is in place and can be used to recreate the scheduler even if it's not used. 

But if you restore and immediately use trainer.test)), then the train ds does not exist (even though the user has already setup test_ds()). This will then raise the above error. In order to avoid raising an error in a valid use case - testing without setting up a training dataset from restoration, this needs to be a warning instead of an error.

Reply:
I though `configure_optimizers` was only called during trainer.fit(), is this wrong?

Reply:
This is right, but PTL actually calls internally trainer.fit() even for trainer.test() (but with different flags preset so as to force eval mode execution)